[knowledge-base] Add backend for feedback page

davidmakoyo opened the pull request:
### Overview
This PR implements a simple feedback system for the knowledge base (see screenshots and more in issue #1004), allowing users to submit general feedback about their experience. 

### Changes Made

1. **Models (`src/chigame/knowledge_base/models.py`)**
   - Added `GeneralFeedback` model with fields:
     - `feedback` (TextField)
     - `created_at` (DateTimeField with auto_now_add)

2. **Views (`src/chigame/knowledge_base/views.py`)**
   - Added `FeedbackView` class that:
     - Handles GET requests to display the form
     - Handles POST requests to save feedback
     - Shows success message after submission

3. **URLs (`src/chigame/knowledge_base/urls.py`)**
   - Added URL pattern: `path("feedback", FeedbackView.as_view(), name="knowledge-base-feedback")`

4. **Templates (`src/templates/knowledge-base/feedback.html`)**
   - Created minimal feedback form with:
     - Simple heading
     - Textarea for feedback
     - Submit button
     - Basic success message

5. **Navigation (`src/templates/knowledge-base/base.html`)**
   - Added "Give Feedback" link in the navbar

### How to Test


1. **Access the Feedback Form**
   - Go to `http://127.0.0.1:8000/knowledge-base/feedback`
   - Or click "Give Feedback" in the knowledge base navbar

2. **Test the Form**
   - Submit empty form (should show validation error)
   - Submit feedback with text (should show success message)
   - Check Django admin panel to verify feedback is saved

3. **Verify in Admin Panel**
   - Go to `http://127.0.0.1:8000/admin`
   - Log in with superuser credentials
   - Check "General feedbacks" section to see submitted feedback

davidmakoyo committed:
[knowledge-base] feedback page backend WIP #1004

davidmakoyo committed:
[knowledge-base] Fix feedback migration conflict and ensure clean migration state

yizou275 reviewed (changes requested):
Thanks David for this work! Yeah it's nice to add a feedback page to knowledge base, and everything works according to your testing instructions :)

I just have a few minor thoughts:
1. The name FeedbackView may look confusing to people unfamiliar with our code base, especially considering we also have a feedback mechanism for guides? 
   - Would it help to maybe rename the view to something like ViewerFeedback/UserFeedback? 
   - Also, maybe add more comment to our views.py, urls.py and other related files, such as clarifying the `FeedbackDetail` view is for displaying the details of a guide, and `FeedbackView` is about displaying the page for user to submit feedback? 

2. Could we add more clarifying descriptions on the page? Like "Pls enter any feedback on the Knowledge Base section of ChiGame."? (This can also avoid user confusion)

3. Have you considered whether to add a `user` field to each piece of feedback object? Now seems that unlogged-in users can also submit feedback -- is this an intentional design? (I personally have no preference, but just asking to check in on what you think)

4. Regarding the success message after user submits the feedback, are you expecting this page? (I'm fine with this as well, but just asking if the form below is supposed to be hided, in your intention.)
![image](https://github.com/user-attachments/assets/d0b586db-7ec8-45d7-b936-e3edc0550942)

Thank you so much!

yizou275 committed:
Merge branch 'dev' into knowledge-base/feedback-backend3

davidmakoyo commented:
Great suggestions thank you Yi!

1. I renamed `FeedbackView` to `UserFeedbackView` and added documentation to clarify the distinction between guide-specific feedback (handled by `FeedbackDetail`) and general user feedback (handled by `UserFeedbackView`), organized the URLs into logical groups with clear comments.

2. Added descriptive title, explanation text, placeholder, and clearer submit button to feedback page.

3. I think that it's fine for non-logged-in users to submit feedback so I'll leave this as is

4. Submitting now redirects to homepage and displays a message 
<img width="1460" alt="image" src="https://github.com/user-attachments/assets/b957c583-ae55-4be9-8156-d94c034e9ac4" />

davidmakoyo committed:
[knowledge-base] improve feedback page closes #1004

davidmakoyo committed:
Merge branch 'dev' into knowledge-base/feedback-backend3

davidmakoyo committed:
[knowledge-base] linter issues and migrations

yizou275 commented:
Thank you so much! Everything looks good to me now. Only one minor issue: your PR seems to include codes from other PR, like moderator publish/unpublish toggle and dark mode. Maybe double check your file changes and undo unrelated changes?

davidmakoyo committed:
Delete src/static/js/knowledge-base/dark-mode-toggle.js

davidmakoyo committed:
Delete src/static/js/knowledge-base/font-size-slider.js

davidmakoyo committed:
Merge branch 'dev' into knowledge-base/feedback-backend3

davidmakoyo commented:
smh no idea how that happened, removed those files

yizou275 reviewed (approved):
Thanks David! This looks good now. I think maybe next time double check `git status` before pushing it and only committing the related files? Since `git add` will add all changed files, including those that you pulled when you are working on the branch. No worries though! :)

jacqsmei reviewed (approved):
nice job! cool feature to have

davidmakoyo committed:
Merge branch 'dev' into knowledge-base/feedback-backend3

davidmakoyo committed:
[knowledge-base] fixed linter errors

jacqsmei commented:
PR Score: Excellent

Comments:
Nice work! Very detailed PR description, super helpful

<!-- {"score": "excellent", "minor": false} -->
